### PR TITLE
Update Webpack => v2

### LINKED ISSUE #3520: Create NPM shrinkwarp file, fix package versions

### PR DESCRIPTION
**Pull request by ngotchac**
This PR is a first WIP for updating UI JS dependencies.

It should fix #3520 as the default saved version of a new installed package begins with `~` (which fixes the version, updating only for patches) instead of the more general `^`.

Webpack is updated to v2, with a working Hot Reload. Eslint is also updated.

This will make updating React and Material UI easier.

### CODE REVIEW

File: js/package.json [hunk 1]
```diff
@@ -26,16 +26,16 @@
   ],
   "scripts": {
     "build": "npm run build:lib && npm run build:dll && npm run build:app",
-    "build:app": "webpack --progress",
-    "build:lib": "webpack --config webpack.libraries --progress",
-    "build:dll": "webpack --config webpack.vendor --progress",
+    "build:app": "webpack --config webpack/config --progress",
```
Comment (jacogr) [inline js/package.json]: Could we maybe reanem the config now? Each are config files, we can call this one `app` (or something similar)

File: js/src/dapps/tokenreg.js [hunk 2]
```diff
@@ -27,13 +27,12 @@ import Container from './tokenreg/Container';
...
 import '../../assets/fonts/RobotoMono/font.css';
 import './style.css';
-import './tokenreg.html';
 
 ReactDOM.render(
   (
-  <Provider store={ store }>
-    <Container />
-  </Provider>
+    <Provider store={ store }>
```
Comment (jacogr) [inline js/src/dapps/tokenreg.js]: Glad this got fixed in eslint, was driving me up the wall.

File: js/src/ui/Form/TypedInput/typedInput.js [hunk 3]
```diff
@@ -22,7 +22,10 @@ import IconButton from 'material-ui/IconButton';
 import AddIcon from 'material-ui/svg-icons/content/add';
 import RemoveIcon from 'material-ui/svg-icons/content/remove';
 
-import { Input, InputAddressSelect, Select } from '../../../ui';
+import Input from '../../../ui/Form/Input';
```
Comment (jacogr) [inline js/src/ui/Form/TypedInput/typedInput.js]: Ahhh, circular detection. Good.

Comment (ngotchac) [inline js/src/ui/Form/TypedInput/typedInput.js]: Still one in `src/api/contract.js` importing `Api` to check that the given `api` in constructor is an instance of `Api` class, and `src/api/api.js` importing `Contract` used in the `newContract` method...

File: js/package.json (hunk 1 above)
Comment (ngotchac) [inline js/package.json]: Done

**Comment by jacogr** [comment]
Merge conflicts.

**Pull request merged**